crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use indexmap::IndexMap;
+use matrix_sdk::room;
+#[cfg(feature = "e2e-encryption")]
+use matrix_sdk::{deserialized_responses::TimelineEvent, Result};
+use ruma::{
+    events::receipt::{Receipt, ReceiptThread, ReceiptType},
+    push::{PushConditionRoomCtx, Ruleset},
+    EventId, OwnedUserId, UserId,
+};
+#[cfg(feature = "e2e-encryption")]
+use ruma::{events::AnySyncTimelineEvent, serde::Raw};
+use tracing::{debug, error};
+
+use super::Profile;
+use crate::timeline::Timeline;
+
+#[async_trait]
+pub trait RoomExt {
+    /// Get a [`Timeline`] for this room.
+    ///
+    /// This offers a higher-level API than event handlers, in treating things
+    /// like edits and reactions as updates of existing items rather than new
+    /// independent events.
+    async fn timeline(&self) -> Timeline;
+}
+
+#[async_trait]
+impl RoomExt for room::Common {
+    async fn timeline(&self) -> Timeline {
+        Timeline::builder(self).track_read_marker_and_receipts().build().await
+    }
+}
```

Comment:
This is likely to be removed once the Sliding Sync FFI API is removed. Thanks for keeping it around for the moment!

Reply:
No, I don't think this one can be removed. This is the most convenient entry point for users of the timeline API that don't use sliding sync, such as Fractal. You probably mean the extension trait from `sliding_sync_ext`, in which case I agree.

Reply:
Ah yeah, I got confused. Sorry for the noise!